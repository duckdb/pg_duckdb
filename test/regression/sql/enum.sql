CREATE TYPE mood AS ENUM (
	'sad',
	'neutral',
	'happy'
);
create table tbl (id serial primary key, a mood);

insert into tbl(a) select 'happy';
insert into tbl(a) select 'neutral';
insert into tbl(a) select 'sad';

select * from tbl;

select * from tbl where a = 'sad'; -- returns `sad`
select * from tbl where a > 'neutral'; --- returns `happy`

ALTER TYPE mood ADD VALUE 'a-bit-happy' BEFORE 'happy';
ALTER TYPE mood ADD VALUE 'very-happy' AFTER 'happy';
ALTER TYPE mood ADD VALUE 'very-sad' BEFORE 'sad';

insert into tbl(a) values ('very-sad'), ('very-happy'), ('a-bit-happy');
select * from tbl where a > 'neutral'; --- returns `a-bit-happy`, `happy` and `very-happy`;
select * from tbl;

drop table tbl;
drop type mood;
