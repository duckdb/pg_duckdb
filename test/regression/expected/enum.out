CREATE TYPE mood AS ENUM (
	'sad',
	'neutral',
	'happy'
);
create table tbl (id serial primary key, a mood);
insert into tbl(a) select 'happy';
insert into tbl(a) select 'neutral';
insert into tbl(a) select 'sad';
select * from tbl;
 id |    a    
----+---------
  1 | happy
  2 | neutral
  3 | sad
(3 rows)

select * from tbl where a = 'sad'; -- returns `sad`
 id |  a  
----+-----
  3 | sad
(1 row)

select * from tbl where a > 'neutral'; --- returns `happy`
 id |   a   
----+-------
  1 | happy
(1 row)

ALTER TYPE mood ADD VALUE 'a-bit-happy' BEFORE 'happy';
ALTER TYPE mood ADD VALUE 'very-happy' AFTER 'happy';
ALTER TYPE mood ADD VALUE 'very-sad' BEFORE 'sad';
insert into tbl(a) values ('very-sad'), ('very-happy'), ('a-bit-happy');
select * from tbl where a > 'neutral'; --- returns `a-bit-happy`, `happy` and `very-happy`;
 id |      a      
----+-------------
  1 | happy
  5 | very-happy
  6 | a-bit-happy
(3 rows)

select * from tbl;
 id |      a      
----+-------------
  1 | happy
  2 | neutral
  3 | sad
  4 | very-sad
  5 | very-happy
  6 | a-bit-happy
(6 rows)

drop table tbl;
drop type mood;
